nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp

```diff
@@ -361,9 +361,64 @@ CComPtr<IAccessible2> getTextBoxInComboBox(
 	return child;
 }
 
+
+std::tuple<long, CComBSTR> getRoleLongRoleString(CComPtr<IAccessible2> pacc, CComVariant varChild) {
+	long role = 0;
+	CComBSTR roleString;
+	CComVariant varRole;
+	if (pacc->role(&role) != S_OK) {
+		role = IA2_ROLE_UNKNOWN;
+	}
+	if (role == 0) {
+		if (pacc->get_accRole(varChild, &varRole) != S_OK) {
+			LOG_DEBUG(L"accRole failed");
+		}
+		if (varRole.vt == VT_I4) {
+			role = varRole.lVal;
+		}
+		else if (varRole.vt == VT_BSTR) {
+			roleString = varRole.bstrVal;
+		}
+	}
+	return std::make_tuple(role, roleString);
+}
+
+
 const vector<wstring>ATTRLIST_ROLES(1, L"IAccessible2::attribute_xml-roles");
 const wregex REGEX_PRESENTATION_ROLE(L"IAccessible2\\\\:\\\\:attribute_xml-roles:.*\\bpresentation\\b.*;");
 
+
+void GeckoVBufBackend_t::fillVBufAriaDetails(
+	int docHandle,
+	CComPtr<IAccessible2> pacc,
+	VBufStorage_buffer_t& buffer,
+	VBufStorage_controlFieldNode_t& parentNode,
+	const std::wstring& roleAttr
+){
+	/* Set the details role by checking for both IA2_RELATION_DETAILS and IA2_RELATION_DETAILS_FOR as one
+	of the nodes in the relationship will not be in the buffer yet */
+	std::optional<int> detailsId = getRelationId(IA2_RELATION_DETAILS, pacc);
+	std::optional<int> detailsForId = getRelationId(IA2_RELATION_DETAILS_FOR, pacc);
+	VBufStorage_controlFieldNode_t* detailsParentNode;

```

Comment:
Don't declare a pointer (or any other POD type) without initializing it. 
>[Read from an indeterminate byte](https://en.cppreference.com/w/cpp/language/default_initialization)
Use of an indeterminate value obtained by default-initializing a non-class variable of any type is [undefined behavior](https://en.cppreference.com/w/cpp/language/ub) (in particular, it may be a [trap representation](https://en.cppreference.com/w/cpp/language/object#Object_representation_and_value_representation)), except in the following cases....

Practically, it may hold a random non zero (therefore not `nullptr`) value, while still being invalid. Causing an error around line 416.

```suggestion
	VBufStorage_controlFieldNode_t* detailsParentNode = nullptr;

```

Reply:
This is picked when you build with `scons source nvdaHelperDebugFlags=analyze`
>`...x86\vbufBackends\gecko_ia2\gecko_ia2.cpp(416) : warning C6001: Using uninitialized memory 'detailsParentNode'.: Lines: 400, 401, 402, 403, 404, 412, 416`